shards/shards.go

```diff
@@ -145,16 +148,60 @@ type shardedSearcher struct {
 
 	rankedVersion uint64
 	ranked        []rankedShard
+
+	priority map[string]float64
 }
 
 func newShardedSearcher(n int64) *shardedSearcher {
 	ss := &shardedSearcher{
-		shards: make(map[string]rankedShard),
-		sched:  newScheduler(n),
+		shards:   make(map[string]rankedShard),
+		sched:    newScheduler(n),
+		priority: make(map[string]float64),
 	}
 	return ss
 }
 
+func (ss *shardedSearcher) watchPriorities(dir string) {
```

Comment:
minor: you don't need to poll, we have infrastructure already for using file watches. But I suppose this is easier / more decoupled which is nice :)

Reply:
Added a comment about it, but an fstat once a minute is quite cheap, and the necessary machinery to setup fsnotify is rather expensive for a single file.